### ISSUE TITLE
Path definitions - software not using constants properly

### ISSUE THREAD

**Issue by ConnectGrid** [opencart/opencart#7938]
**What version of OpenCart are you reporting this for?**
3.0

**Describe the bug**
The front controller defines constants to represent various paths used by OpenCart, but then these constants are not being used throughout the code, particularly during the install process.

For example, DIR_IMAGE is defined as the path to the /image/ directory, but this constant is not being used. Instead, the install steps all assume that the image directory is in a particular location and simply recompose that assumed path.

_(Our particular needs do not use the default OpenCart locations. Even though we correctly updated the front controller to define the correct paths, the rest of the install scripts ignore the constants.)_

**To Reproduce**
1. Change the path of a path constant in the front controller (index.php)
2. Attempt to install
3. Any reference to that path is different from what is defined in index.php.

**Expected behavior**
The code correctly uses the defined constants. That's what constants are for... define once and use everywhere safely.

**Screenshots / Screen recordings**
n/a

**Server / Test environment (please complete the following information):**
 - development
 - macOS
 - Nginx / PHP-FPM 7.4 (fully updated)

**Additional context**
n/a

**Comment by ConnectGrid** [comment]
If we submit a pull request that correctly utilizes path constants throughout the code, will this pull request be given consideration?

As part of this, we'd suggest that the path constants not be prefixed with `DIR_`, as this assumes that OpenCart is being used as the only software, not co-existing with other software. Our particular needs OpenCart to co-exist, so we suggest using an `OPENCART_` prefix instead. This gives proper scope and is self-explanatory.

Examples:
- `DIR_SYSTEM` becomes `OPENCART_SYSTEM`
- `DIR_CACHE` becomes `OPENCART_CACHE`
- `DIR_IMAGE` becomes `OPENCART_IMAGE`
etc.

This way the constants have no way of conflicting with other constants that may be defined by other software/components.

**Comment by straightlight** [comment]
The only problematic by doing that is the mount of extensions that may have their own defined constants as well as opposed to the OPENCART_ prefixes. Those can be pretty easy to track but all it would take, unfortunately, is one extension developer who would define their own OPENCART_ and store owners requesting the need to troubleshoot might believe it's a supported issue while it may not be either on the OpenCart forum or on this namespace.

**Comment by ConnectGrid** [comment]
@straightlight That would be no different than an extension developer defining a constant prefixed with "DIR_" or anything else.

I think I sidetracked this issue by discussing the naming convention. Back on track...

Any thoughts on why install/index.php defines these DIR_* constants, but then doesn't use them during the installation? Was it a work-in-progress?

The need is to have one place to define paths, not several places. Would you welcome a pull request that fixes this?

**Comment by straightlight** [comment]
```That would be no different than an extension developer defining a constant prefixed with "DIR_" or anything else.```

Then, this simply means that it still won't prevent extension developers to use the DIR_ constants neither. 

```Any thoughts on why install/index.php defines these DIR_* constants, but then doesn't use them during the installation? Was it a work-in-progress?```

Simple. The master branch is always a work-in-progress process. It's rarely a definite thing to show from a Github namespace.

```The need is to have one place to define paths, not several places.```

The only places required to define constant paths from the core is in config.php and admin/config.php files. I wouldn't consider those two as several places. In fact, they're pretty simply to change as it only requires 2 files to do it. When it comes to extensions, however, depending on the extensions being developed and used, while more constants could be involved, they're not supported by Opencart originally on anyhow.

**Comment by ConnectGrid** [comment]
Yes, I finally figured this all out.

- During the install, even though install/index.pho defines the DIR_ constants, these are _not_ correctly/fully used in the rest of the step2/3 scripts.
- The installer _generates_ config.php files for the store, which is where the constants are defined. I was not aware of this at first.
- (The master branch threw me for a loop with regards to dependencies. I needed to be on the 3.0.x.x.Maintenance branch, which had only added to my initial confusion.)

Conclusion - the installer does need some cleanup, but I'm guessing this is low priority since it's generally a one-time run and works as designed right now, so no "fixing" needed.

* I did do this cleanup in my fork, reducing the code by a notable amount (less duplication), and I'll submit a pull request for consideration.
